iabtcf-encoder/src/main/java/com/iabtcf/encoder/BitFieldEncoder.java

```diff
@@ -0,0 +1,166 @@
+package com.iabtcf.encoder;
+
+/*-
+ * #%L
+ * IAB TCF Java Encoder Library
+ * %%
+ * Copyright (C) 2020 IAB Technology Laboratory, Inc
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.BitSet;
+import java.util.Iterator;
+
+import com.iabtcf.FieldDefs;
+import com.iabtcf.utils.IntIterable;
+import com.iabtcf.utils.IntIterator;
+
+/**
+ * Encodes a bit / range field, typically used for encoding vendor lists.
+ */
+class BitFieldEncoder {
+    private final BitSet vendors;
+    private int maxVendorId;
+
+    public BitFieldEncoder() {
+        this(new BitSet(), -1);
+    }
+
+    public BitFieldEncoder(BitFieldEncoder prototype) {
+        this(prototype.vendors.length() == 0 ? new BitSet() : prototype.vendors.get(0, prototype.vendors.length()),
+                prototype.maxVendorId);
+    }
+
+    private BitFieldEncoder(BitSet vendors, int maxVendorId) {
+        this.vendors = vendors;
+        this.maxVendorId = maxVendorId;
+    }
+
+    public BitFieldEncoder add(int vendorId) {
+        vendors.set(vendorId - 1);
```

Comment:
I think a 0 default for max makes more sense, this how it comes when we the following list is empty. We dont support negative numbers anywhere

Reply:
i like this.. and also the zero value can still support the ability to set a max vendor id greater than the max(vendorIds).